src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -560,16 +560,6 @@ internal override bool IsRecord
         {
             get
             {
-                // First, do a check to see if there are even any members named <Clone>$. If we have to actually find the clone
-                // method, it can be expensive, because we have to load all the members. MemberNames just loads the strings, which
-                // is much cheaper.
-                if (!MemberNames.Contains(WellKnownMemberNames.CloneMethodName))
-                {
-                    return false;
-                }
-
-                // There exists a member named <Clone>$. We still need to check to see if it's actually a valid clone method, but
-                // there's no getting around that now.
```

Comment:
Should we `Debug.Assert(HasPossibleWellKnownClonMethod())` here? It seems any time that assert tripped would be a performance issue we'd want to dig into.

Reply:
We can't. That check was moved into `FindValidCloneMethod` itself.